### DISCUSSION TITLE
[Help] Is there a way to enable line numbers in the chat buffer?

### DISCUSSION THREAD

**Discussion by Ajaymamtora** [olimorris/codecompanion.nvim#526] [Q&A] [answered]
As the title says, I've tried:

```lua
-- opts.display.chat.window.opts...
              relativenumber = true,
```

And it doesn't work.

**Comment by bassamsdata** [comment]
> -- opts.display.chat.window.opts...


I'm not sure about this but you can do it in a file `nvim/ftplugin/codecompanion.lua`

```lua
vim.schedule(function() -- `schedule` is required to work
  vim.opt_local.relativenumber = true
end)
```

**Reply by Ajaymamtora** [reply]
Doesnt work for me unfortunately, thanks tho

**Reply by bassamsdata** [reply]
Oh, I must have misunderstood your question. Are you asking about displaying relative numbers with showing the current line number? If so, you'll need:
```lua
-- opts.display.chat.window.opts
number = true,
relativenumber = true
```

**Reply by tristan957** [reply]
Why does it have to be wrapped in a `vim.schedule()`? That isn't the case for other ftplugin files that I have.

**Reply by bassamsdata** [reply]
I'm not entirely sure, as it was a long time ago, but it was likely due to the execution timeline, we needed to set it after initializing the chat buffer.  

That said, it might not be necessary. You can probably configure it properly by setting the option directly in the plugin's configuration [here](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua#L926).

**Reply by tristan957** [reply]
Unfortunately it is still necessary. I need the ftplugin because I need to set some other variables too.

**Reply by sghng** [reply]
> > -- opts.display.chat.window.opts...
> 
> I'm not sure about this but you can do it in a file `nvim/ftplugin/codecompanion.lua`
> 
> ```lua
> vim.schedule(function() -- `schedule` is required to work
>   vim.opt_local.relativenumber = true
> end)
> ```

@bassamsdata Is there any reason that it must be scheduled? I also noticed that any `set` operations don't work in `after/ftplugin/codecompanion.vim`

**Comment by sghng** [comment]
Another thing I want to point out is seems like a lot of settings in `after/ftplugin/codecompanion.{vim,lua}` are not respected. Specifically, I found key mappings are effective, but anything with `set` are ignored. (I'm using vim script for this)

**Comment by olimorris** [comment, marked as answer]
This has now been fixed in #2251 . Huge shout out to @bassamsdata for this fix!
